### ISSUE TITLE
04_Deconv.cntk Sample not converging?

### ISSUE THREAD

**Issue by kestrelm** [Microsoft/CNTK#749]
Hello,

I am running the 04_Deconv.cntk sample with CNTK GPU binary on a Nvidia GTX 950 card.

Even at epoch 168, the mse stays at around 41. Is this expected behaviour? Or is there something else I am missing here. Should the mean square error drop a lot more after > 100 epochs? 

Here are the printouts:

> Starting minibatch loop.
>  Epoch[165 of 1000]-Minibatch[   1- 500, 26.67%]: mse = 41.67640234 \* 16000; time = 0.5849s; samplesPerSecond = 27356.9
>  Epoch[165 of 1000]-Minibatch[ 501-1000, 53.33%]: mse = 41.45493359 \* 16000; time = 0.5782s; samplesPerSecond = 27674.1
>  Epoch[165 of 1000]-Minibatch[1001-1500, 80.00%]: mse = 41.60703906 \* 16000; time = 0.5814s; samplesPerSecond = 27519.8
> Finished Epoch[165 of 1000]: [Training] mse = 41.56842917 \* 60000; totalSamplesSeen = 9900000; learningRatePerSample = 3
> .1250001e-005; epochTime=2.18283s
> SGD: Saving checkpoint model '../Output/Models/04_DeConv.165'
> 
> Starting Epoch 166: learning rate per sample = 0.000031  effective momentum = 0.900000  momentum as time constant = 303.
> 7 samples
> BlockRandomizer::StartEpoch: epoch 165: frames [9900000..9960000](first sequence at sample 9900000), data subset 0 of 1
> 
> Starting minibatch loop.
>  Epoch[166 of 1000]-Minibatch[   1- 500, 26.67%]: mse = 41.61532422 \* 16000; time = 0.5793s; samplesPerSecond = 27620.3
>  Epoch[166 of 1000]-Minibatch[ 501-1000, 53.33%]: mse = 41.52181641 \* 16000; time = 0.5902s; samplesPerSecond = 27110.4
>  Epoch[166 of 1000]-Minibatch[1001-1500, 80.00%]: mse = 41.36217969 \* 16000; time = 0.5713s; samplesPerSecond = 28004.8
> Finished Epoch[166 of 1000]: [Training] mse = 41.46359583 \* 60000; totalSamplesSeen = 9960000; learningRatePerSample = 3
> .1250001e-005; epochTime=2.18893s
> SGD: Saving checkpoint model '../Output/Models/04_DeConv.166'
> 
> Starting Epoch 167: learning rate per sample = 0.000031  effective momentum = 0.900000  momentum as time constant = 303.
> 7 samples
> BlockRandomizer::StartEpoch: epoch 166: frames [9960000..10020000](first sequence at sample 9960000), data subset 0 of
> 1
> 
> Starting minibatch loop.
>  Epoch[167 of 1000]-Minibatch[   1- 500, 26.67%]: mse = 41.47810547 \* 16000; time = 0.5815s; samplesPerSecond = 27512.8
>  Epoch[167 of 1000]-Minibatch[ 501-1000, 53.33%]: mse = 41.42062891 \* 16000; time = 0.5846s; samplesPerSecond = 27371.0
>  Epoch[167 of 1000]-Minibatch[1001-1500, 80.00%]: mse = 41.21780469 \* 16000; time = 0.5689s; samplesPerSecond = 28123.1
> Finished Epoch[167 of 1000]: [Training] mse = 41.35938333 \* 60000; totalSamplesSeen = 10020000; learningRatePerSample =
> 3.1250001e-005; epochTime=2.17677s
> SGD: Saving checkpoint model '../Output/Models/04_DeConv.167'
> 
> Starting Epoch 168: learning rate per sample = 0.000031  effective momentum = 0.900000  momentum as time constant = 303.
> 7 samples
> BlockRandomizer::StartEpoch: epoch 167: frames [10020000..10080000](first sequence at sample 10020000), data subset 0 o
> f 1
> 
> Starting minibatch loop.
>  Epoch[168 of 1000]-Minibatch[   1- 500, 26.67%]: mse = 41.32962109 \* 16000; time = 0.5783s; samplesPerSecond = 27667.2
>  Epoch[168 of 1000]-Minibatch[ 501-1000, 53.33%]: mse = 41.31296484 \* 16000; time = 0.5733s; samplesPerSecond = 27906.7
>  Epoch[168 of 1000]-Minibatch[1001-1500, 80.00%]: mse = 41.21615625 \* 16000; time = 0.5803s; samplesPerSecond = 27570.5
> Finished Epoch[168 of 1000]: [Training] mse = 41.25573750 \* 60000; totalSamplesSeen = 10080000; learningRatePerSample =
> 3.1250001e-005; epochTime=2.17607s
> SGD: Saving checkpoint model '../Output/Models/04_DeConv.168'

**Comment by pkranen** [comment]
We have a new example for deconvolution for both Python and BrainScript. Please check these links:

[Examples/Image/GettingStarted/07_Deconvolution_PY.py](https://github.com/Microsoft/CNTK/blob/master/Examples/Image/GettingStarted/07_Deconvolution_PY.py)

[https://github.com/Microsoft/CNTK/wiki/Image-Auto-Encoder-Using-Deconvolution-And-Unpooling](https://github.com/Microsoft/CNTK/wiki/Image-Auto-Encoder-Using-Deconvolution-And-Unpooling)

I close this issue, feel free to reopen.
